redfish-core/lib/redfish_sessions.hpp

```diff
@@ -197,6 +197,63 @@ inline void handleSessionCollectionMembersGet(
     asyncResp->res.jsonValue = getSessionCollectionMembers();
 }
 
+inline void processAfterSessionCreation(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const crow::Request& req,
+    std::shared_ptr<persistent_data::UserSession> session,
+    bool isGenerateSecretkeyRequired)
+{
+    // When session is created by webui-vue give it session cookies as a
+    // non-standard Redfish extension. This is needed for authentication for
+    // WebSockets-based functionality.
+    if (!req.getHeaderValue("X-Requested-With").empty())
+    {
+        bmcweb::setSessionCookies(asyncResp->res, *session);
+    }
+    else
+    {
+        asyncResp->res.addHeader("X-Auth-Token", session->sessionToken);
+    }
+
+    boost::urls::url sessionUrl = boost::urls::format(
+        "/redfish/v1/SessionService/Sessions/{}", session->uniqueId);
+    asyncResp->res.addHeader("Location", sessionUrl.buffer());
+    asyncResp->res.result(boost::beast::http::status::created);
+    if (session->isConfigureSelfOnly)
+    {
+        auto accountUri = boost::urls::format(
+            "/redfish/v1/AccountService/Accounts/{}", session->username);
+        if (!isGenerateSecretkeyRequired)
+        {
+            messages::passwordChangeRequired(asyncResp->res, accountUri);
+        }
+        else
+        {
+            messages::generateSecretKeyRequired(asyncResp->res, accountUri);
+        }
+    }
+
+    crow::getUserInfo(asyncResp, session->username, session,
+                      [asyncResp, session]() {
+        fillSessionObject(asyncResp->res, *session);
+    });
+}
+
+inline void checkGoogleAuthenticatorSecretKeyRequired(
+    const std::string& username,
+    std::function<void(const boost::system::error_code& ec, bool)> callback)
+{
+    sdbusplus::message::object_path tempObjPath("/xyz/openbmc_project/user");
+    tempObjPath /= username;
+    const std::string userPath(tempObjPath);
```

Comment:
why do you need the std::string 
https://github.com/openbmc/bmcweb/blob/463a0e3e880340a119bf986bab039d1afbe1a432/redfish-core/lib/certificate_service.hpp#L400 

ABove is code calling 
crow::connections::systemBus->async_method_call
with sdbusplus::message::object_path

Reply:
userPath is a better name than tempObjPath so keep that name

Reply:
Updated